### ISSUE TITLE
POSIX compliant shell

### ISSUE THREAD

**Issue by Vanuan** [vishnubob/wait-for-it#8]
Is it possible to make this shell-independent?

    /wait-for-it.sh: line 70: syntax error: unexpected "(" (expecting ";;")

This would be particularly useful to run it with alpine, which uses busybox.

**Comment by Vanuan** [comment]
@iturgeon have you looked into this too?

**Comment by iturgeon** [comment]
Strange, I don't think my alpine 3.3 container complained about this line.

**Comment by Vanuan** [comment]
Yes, I've added this too as a workaround.

**Comment by scarlac** [comment]
To add to @iturgeon's comment, 
I also fixed this issue by replacing `sh /code/wait-for-it.sh ....` with `bash /code/wait-for-it.sh ...`

**Comment by maxcnunes** [comment]
Since this issue is open for while. Let me share a tool I have written in Go with cross platform support and which has the same purpose of this project (even the name is the same) https://github.com/maxcnunes/waitforit.

**Comment by Vanuan** [comment]
What's great in shell script is that you just add one file to your directory and if you have bash that's it. Since your script is written in go, and go is a part of docker, it can be executed without a shell, right?

**Comment by maxcnunes** [comment]
For UNIX systems the approach of this project is really straight forward. But it may start to be a bit complicated to maintain this approach once:
* need to add support for different OS and platforms
* need to add new features such as supporting http code checking (may would depend on third party tools such as curl)
An approach in Go solve those problems because it generates a binary. So to add support for a new OS/platform would only be required to build a binary for that target. And adding new features is simple because in most cases would require using a Go built in package or maybe another external package. But that would not be a problem for different OS/platform because of the binary.
*PS:* The built binary does not need Go installed in the machine to be able to run. Is a totally self contained executable.

**Comment by Vanuan** [comment]
I meant you can build it using a Dockerfile, without the need to download, right?

**Comment by Vanuan** [comment]
It would be great if it was shipped with alpine.

**Comment by Vanuan** [comment]
OS support is irrelevant, since bash would be supported in windows and Docker doesn't currently support any other platform except linux (windows 2016 is still not released).

**Comment by maxcnunes** [comment]
I don't think so you have access to Go from the Dockerfile. Because the Docker running the build on the Dockerfile is already a built binary. Does not have all the Golang support to build a new binary.
OS support is relevante if you think it depend on third party tools. For instance bash itself is not available in all images by default. You would still need to install that at least. And as I said before, if it start to include new features such as the http status checking, it may start to depend on other tools as well. Which means, you have to include in your image more things than a single binary.

But if the these points I am bringing up does not concern you and the solution of this project works for you. Please than just use this one. It is a small and straight forward solution. 😄

**Comment by Vanuan** [comment]
I'm open to new solution. It's just that while it solves one problem (not having to have bash) it adds another (having to build or download a binary)

**Comment by maxcnunes** [comment]
@Vanuan Just a reminder. If you don't want to your Dockefile be responsible to download the binary. You could also use [`COPY` do add the binary from your file system](https://github.com/maxcnunes/waitforit#using-copy-from-local-file-system).
Actually even with this tool you would end up with the same "problem": download or copy the script from the file system.

**Comment by Vanuan** [comment]
Sorry, I never commit binaries to a git repository

**Comment by Vanuan** [comment]
@maxcnunes please, don't be defensive, I think your project is great! I'm just sharing my thoughts

**Comment by maxcnunes** [comment]
I'm not 😄  Just sharing some thoughts as well.

**Comment by aviau** [comment]
You guys are hijacking this thread, please stop.

**Comment by dborncamp** [comment]
Has anyone found a workaround for this? I just hit it in a `python:3.7.2-alpine3.9` image.

**Comment by ChadLei** [comment]
im also getting this with a openjdk:8-jre manifold image

**Comment by qg0** [comment]
This doesn't work with classical shell (**sh**).

Docker image: php:fpm-alpine
Shell version BusyBox v1.31.1 () multi-call binary.

line 75: syntax error: unexpected "(" (expecting ";;")
 `WAITFORIT_hostport=(${1//:/ })`

line 121: syntax error: unexpected "(" (expecting ";;")
 `WAITFORIT_CLI=("$@")`

line 133: syntax error: bad substitution
```
    esac
done
//line 133
if [[ "$WAITFORIT_HOST" == "" || "$WAITFORIT_PORT" == "" ]]; then
    echoerr "Error: you need to provide a host and port to test."
    usage
fi
```
